File: MEMBERSHIP/MEMBERSHIP.UI/src/models/auth/membersDto.ts
export interface IMembersPostDto {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  email: string;
  Zone: string;
  RegionId: string;
  woreda: string;
  inistitute: string;
  membershipTypeId: string;
}

export interface IMembersGetDto {

  id: string;
  fullName: string;
  imagePath: string;
  phoneNumber: string;
  email: string;
  region: string;
  zone: string;
  woreda: string;
  inistitute: string;
  birthDate: string;
  membershipTypeId: string;
  educationalField: string;
  educationalLevel: string;
  lastPaid: Date;
  educationalLevelId: string;
  membershipType: string;
  memberId: string;
  gender: string;
  instituteRole: string;
  amount: number;
  text_Rn: string;
  expiredDate: Date;
  paymentStatus: string;
  idCardStatus: string;
  rejectedRemark: string;
  isBirthDate: boolean;
  moodleName: string;
  moodlePassword?: string;
  moodleId: string;
  moodleStatus: string;
  createdByDate: Date;
  currency: string;
  regionId?: string;
  memberStatus?:string;
  membershipCategory?: string;
}
export interface ICompletePorfileDto {
  id: string;
  educationalField: string;
  educationalLevelId: string;
  gender: string;
  instituteRole: string;
  birthDate: Date;
}

export interface IMemberUpdateDto {
  id: string;
  fullName: string;
  email?: string;
  phoneNumber: string;
  educationalField: string;
  educationalLevelId: string;
  gender: string;
  institute: string;
  birthDate: Date;
  woreda: string;
  instituteRole: string;
  regionId? : string

  lastPaid?: Date;
  expiredDate?: Date;
  paymentStatus?: string;

  membershipTypeId?: string;
}

export interface IMemberTelegramDto {
  id: string;
  fullName: string;
  email?: string;
  phoneNumber: string;
  memberId: string;
  amount: number;
  text_Rn: string;
  expiredDate: Date;
  paymentStatus: Date;
  membershipType: string;
  membershipTypeId: string;
  url: string;
  currency: string;
}

export interface MoodleUpdateDto {
  memberId: string;
  moodleName: string;
  moodlePassword?: string;
  moodleId: string;
}

File: MEMBERSHIP/MEMBERSHIP.API/MembershipImplementation/DTOS/Configuration/MembersDto.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MembershipImplementation.DTOS.Configuration
{
    public record MembersPostDto
    {
        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string PhoneNumber { get; set; }

        public string Email { get; set; }

        public Guid? ZoneId { get; set; }

        public string? Woreda { get; set; }

        public string Inistitute { get; set; }

        public Guid MembershipTypeId { get; set; }
        

    }

    public record  MemberUpdateDto
    {

        public Guid Id { get; set; }
        public string FullName { get; set; }
        public DateTime BirthDate { get; set; }
        public string PhoneNumber { get; set; }
        public string EducationalField { get; set; }
        public Guid EducationalLevelId { get; set; }
        public string Gender { get; set; }

        public IFormFile? Image { get; set; }
        public string Institute { get; set; }

        public string? Email { get; set; }

        public string? Woreda { get; set; }

        public string InstituteRole { get; set; }
    }


    


    public record MembersGetDto
    {
        public string Id { get; set; }
        public string FullName { get; set; }
        public string ImagePath { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public DateTime BirthDate { get; set; }
        public string Region { get; set; }
        public string Zone { get; set; }
        public string Woreda { get; set; }
        public string Inistitute { get; set; }
        public bool IsBirthDate { get; set; }


        public string MembershipTypeId { get; set; }
        public string EducationalField { get; set; }
        public string EducationalLevel { get; set; }


        public string EducationalLevelId { get; set; }
        public string MembershipType { get; set; }
        public string MemberId { get; set; }
        public string Gender { get; set; }
        public string InstituteRole { get; set; }

        public double Amount { get; set; }

        public string Text_Rn { get; set; }

        public DateTime ExpiredDate { get; set; }

        public string PaymentStatus { get; set;  }

        public string IdCardStatus { get; set; }

        public string RejectedRemark { get; set; }

        public bool IsProfileCompleted { get; set; }


        public string? MoodleName { get; set; }

        public string? MoodlePassword { get; set;}

        public string? MoodleId { get;set; }

        public string? MoodleStatus { get; set; }

    }


    public class MemberTelegramDto
    {
        public Guid Id { get; set; }
        public string FullName { get; set; }       
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string MemberId { get; set; }
        public double Amount { get; set; }
        public string Text_Rn { get; set; }
        public DateTime ExpiredDate { get; set; }
        public string PaymentStatus { get; set; }

        public string Url { get; set; }
        public string MembershipType { get; set; }

        public string MembershipTypeId { get; set; }

    }
    public class CompleteProfileDto
    {
        public Guid Id { get; set; }
        public string EducationalField { get; set; }
        public string EducationalLevelId { get; set; }
        public string Gender { get; set; }
        public string InstituteRole { get; set; }

        public DateTime BirthDate { get; set; }

        public IFormFile Image { get; set; }
    }
}
